Sources/DistributedCluster/LifecycleMonitoring/LifecycleWatchContainer.swift

```diff
@@ -218,16 +238,6 @@ extension LifecycleWatchContainer {
                 continue
             }
 
```

Comment:
I think we need locking in this method too, for `self.watching` on L216/236.

Reply:
Right yeah and a few more, I'm cleaning up this file some more still. thank you for review!